bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -976,13 +1089,41 @@ void sendAddSuccessCallbacks() {
                 LOG.debug("Head of the queue entryId: {} is not lac: {} + 1", pendingAddOp.entryId, lastAddConfirmed);
                 return;
             }
+
             pendingAddOps.remove();
+            piggyBackedLastAddConfirmed = lastAddConfirmed;
             lastAddConfirmed = pendingAddOp.entryId;
+
             pendingAddOp.submitCallback(BKException.Code.OK);
         }
 
     }
 
+    private void explictLastAddConfirmedUpdate(final LedgerHandle lh) {
+        int explicitLacIntervalInSec = bk.getExplicitLacInterval();
+        final Runnable updateLacTask = new Runnable() {
+            @Override
+            public void run() {
+                if (lh.getExplicitLastAddConfirmed() < lh.getPiggyBackedLastAddConfirmed()) {
+                    LOG.debug("ledgerid: {}", lh.getId() );
+                    LOG.debug("explicitLac:{} piggybackLac:{}", lh.getExplicitLastAddConfirmed(), lh.getPiggyBackedLastAddConfirmed() );
+                    lh.setExplicitLastAddConfirmed(lh.getPiggyBackedLastAddConfirmed());
+                    return;
+                }
+
+                // explicitLastAddConfirmed >= piggybackedLastAddConfirmed
+                if (lh.getLastAddConfirmed() > lh.getExplicitLastAddConfirmed()) {
+                    // Send Explicit LAC
+                    LOG.debug("ledgerid: {}", lh.getId() );
+                    asyncLastAddConfirmedUpdate(lh.getLastAddConfirmed());
+                    lh.setExplicitLastAddConfirmed(lh.getLastAddConfirmed());
+                    LOG.debug("After sending explict LAC lac: {}  explicitLac:{}", lh.getLastAddConfirmed(), lh.getExplicitLastAddConfirmed());
+                }
+            }
+        };
+        this.lastAddConfirmedUpdateScheduler.scheduleAtFixedRate(updateLacTask, explicitLacIntervalInSec, explicitLacIntervalInSec, SECONDS);
```

Comment:
This flush here will do unnecessary flushes here.

A better algorithm here is:

- you should track the last piggyback lac update time.
- when the scheduled task is running, it should compare the last piggyback lac update time to see if piggyback lac update already elapsed more than the interval. if it is more than the interval, add the lac explicitly again.